src/read_utils.jl

```diff
@@ -2,6 +2,37 @@ using DelimitedFiles
 
 # ---------------------------------------------------------------------------------------- #
 
+function make_symmetryvector(bandirsd::Dict{String, Vector{Pair{UnitRange{Int64}, Vector{Int64}}}}, 
+lgirsd::Dict{String, Vector{LGIrrep{2}}}, sgnum::Integer, dim::Integer=2, has_tr::Bool = true)
+    sb, _ = compatibility_basis(sgnum, dim; timereversal=has_tr)
```

Comment:
I think it would be quite bad to have this call `compatibility_basis` for every dataset: the reason being that `compatibility_basis` can be quite slow in some space groups (crazy integer-programming problems being solved).

We basically only need `sb` here in order to construct the permutation vector, right? If so, maybe we could just say that the method should be *passed* a `BandRepSet` (it has the same sorting as `sb`). In fact, I think it would be better if this method accepted a permutation vector `permd` - and if it isn't supplied, it would be calculated based on a kwarg input of a `BandRepSet`?

Reply:
This sounds like a better implementation- I'll alter the pull request now